components/raftstore/src/coprocessor/region_info_accessor.rs

```diff
@@ -450,6 +462,19 @@ impl RegionCollector {
         callback(&mut iter)
     }
 
+    pub fn handle_find_region_by_key(&self, key: Vec<u8>, callback: Callback<Option<Region>>) {
+        let key = RangeKey::from_start_key(key);
+        let mut region = None;
+        if let Some((_, region_id)) = self.region_ranges.range((Excluded(key), Unbounded)).next() {
+            region = self
+                .regions
+                .get(region_id)
+                .map(|region_info| region_info.region.clone());
```

Comment:
I think you need to check if the region contains the specified key. If the region that contains the key is not present in the collection, it seems that the first region that has keys greater than the given key will be found, which I think is not what you expect.

Reply:
Good catch.